pkg/sandbox/sandbox.go

```diff
@@ -78,6 +101,110 @@ func (s *sandbox) SetConfig(config *types.PodSandboxConfig) error {
 	return nil
 }
 
+// returns the spec
+func (s *sandbox) Spec() *generate.Generator {
+	return &s.spec
+}
+
+func (s *sandbox) SpecAddAnnotations(pauseImage, containerName, shmPath, privileged, runtimeHandler, resolvPath, hostname, stopSignal, cgroupParent, mountPoint, hostnamePath, cniResultJSON, created string, ips []string, isSystemd, spoofedContainer bool) (err error) {
+	kubeAnnotations := s.config.Annotations
+	// add metadata
+	metadata := s.config.Metadata
+	metadataJSON, err := json.Marshal(metadata)
+	if err != nil {
+		return err
+	}
+
+	labels := s.config.Labels
+
+	// Add special container name label for the infra container
+	if s.config.Labels != nil {
+		labels[kubeletTypes.KubernetesContainerNameLabel] = leaky.PodInfraContainerName
+	}
+	labelsJSON, err := json.Marshal(labels)
+	if err != nil {
+		return err
+	}
+
+	// add annotations
+	kubeAnnotationsJSON, err := json.Marshal(s.config.Annotations)
+	if err != nil {
+		return err
+	}
+
+	securityContext := s.config.Linux.SecurityContext
+	nsOptsJSON, err := json.Marshal(securityContext.NamespaceOptions)
+	if err != nil {
+		return err
+	}
+
+	logDir := s.config.LogDirectory
+	logPath := filepath.Join(logDir, s.ID()+".log")
+	kubeName := s.config.Metadata.Name
+
+	hostNetwork := securityContext.NamespaceOptions.Network == types.NamespaceModeNODE
+
+	s.spec.AddAnnotation(annotations.Metadata, string(metadataJSON))
+	s.spec.AddAnnotation(annotations.Labels, string(labelsJSON))
+	s.spec.AddAnnotation(annotations.Annotations, string(kubeAnnotationsJSON))
+	s.spec.AddAnnotation(annotations.LogPath, logPath)
+	s.spec.AddAnnotation(annotations.Name, s.Name())
+	s.spec.AddAnnotation(annotations.Namespace, s.config.Metadata.Namespace)
+	s.spec.AddAnnotation(annotations.ContainerType, annotations.ContainerTypeSandbox)
+	s.spec.AddAnnotation(annotations.SandboxID, s.ID())
+	s.spec.AddAnnotation(annotations.Image, pauseImage)
+	s.spec.AddAnnotation(annotations.ContainerName, containerName)
+	s.spec.AddAnnotation(annotations.ContainerID, s.ID())
+	s.spec.AddAnnotation(annotations.ShmPath, shmPath)
+	s.spec.AddAnnotation(annotations.PrivilegedRuntime, fmt.Sprintf("%v", privileged))
+	s.spec.AddAnnotation(annotations.RuntimeHandler, runtimeHandler)
+	s.spec.AddAnnotation(annotations.ResolvPath, resolvPath)
+	s.spec.AddAnnotation(annotations.HostName, hostname)
+	s.spec.AddAnnotation(annotations.NamespaceOptions, string(nsOptsJSON))
+	s.spec.AddAnnotation(annotations.KubeName, kubeName)
+	s.spec.AddAnnotation(annotations.HostNetwork, fmt.Sprintf("%v", hostNetwork))
+	s.spec.AddAnnotation(annotations.ContainerManager, lib.ContainerManagerCRIO)
+	s.spec.AddAnnotation(annotations.Created, created)
```

Comment:
is created used in runPodSandbox at all? if now, we can just call time.Now() here and drop the arg

Reply:
I tried doing that but it was causing errors because there was a discrepancy between when the pod was "created" and when it set the annotation